### PR TITLE
Uml 881 expire dialog

### PR DESCRIPTION
**Pull request by johnnolan**
## Purpose
Creates a dialog box in JS ready for the implementation of the session timeout. 

Fixes UML-881

## Approach

Adds in a timeout and a dialog for when when a session is about to expire.

## Learning

Cannot test key pressses in JSDom which means our testing is a little short on the JS. One to investigate in the future or cover via Behat in the future.

## Checklist

* [ ] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [x] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-front/web/src/javascript/sessionDialog.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        this.element = element;
+        this.countdownMinutes = countdownMinutes; // Temporary until final solution in place
+        this.dialogOverlay = document.getElementById("dialog-overlay");
+        this.dialogFocus = document.querySelector(".dialog-focus");
+        this._setupEventHandlers();
+        this._trapFocus();
+        this.timer = new countdownTimer(this.element.querySelector('#time'), this.countdownMinutes);
+        this.timer.on('tick', (event) => {
+            if (event === 5) {
```
Comment (williamfalconeruk) [inline service-front/web/src/javascript/sessionDialog.js:14]: could we make it clearer what the `event`value of 5 is in a `const` or something similar?

Comment (johnnolan) [reply service-front/web/src/javascript/sessionDialog.js]: Good call, have updated this 👍

File: service-front/web/src/javascript/sessionDialog.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import countdownTimer from "./countdownTimer";
+
+export default class SessionDialog {
+    constructor(element, countdownMinutes)
+    {
+        this.showDialogMinutesBeforeLogout = 5;
```
Comment (williamfalconeruk) [inline service-front/web/src/javascript/sessionDialog.js:6]: I guess this value will need to be configurable when we plumb this dialog in?

Comment (johnnolan) [reply service-front/web/src/javascript/sessionDialog.js:6]: Yeah that sounds like a good idea, also the countdown time as well would be nice to configure this way!

**Pull request merged**